ruby/lib/ci/queue/redis/build_record.rb

```diff
@@ -59,20 +59,10 @@ def record_warning(type, attributes)
         Test = Struct.new(:id) # Hack
 
         def record_error(id, payload, stats: nil)
-          # FIXME: the ack and hset should be atomic
-          # otherwise the reporter may see the empty queue before the error
-          # is appended and wrongly think it's a success.
-          if @queue.acknowledge(id)
-            redis.pipelined do |pipeline|
-              pipeline.hset(
-                key('error-reports'),
-                id,
-                payload,
-              )
-              pipeline.expire(key('error-reports'), config.redis_ttl)
-              record_stats(stats, pipeline: pipeline)
-              @queue.increment_test_failed
-            end
+          @queue.acknowledge(id, error: payload)
+          redis.pipelined do |pipeline|
+            record_stats(stats, pipeline: pipeline)
+            @queue.increment_test_failed
           end
```

Comment:
```suggestion
          redis.pipelined do |pipeline|
            @queue.acknowledge(id, error: payload)
            record_stats(stats, pipeline: pipeline)
            @queue.increment_test_failed
          end
```

If the return value don't matter, you can pipeline it, no?